consensus/src/chained_bft/chained_bft_smr.rs

```diff
@@ -183,29 +180,51 @@ impl<T: Payload, P: ProposerInfo> ChainedBftSMR<T, P> {
         executor: TaskExecutor,
         mut receiver: channel::Receiver<ProposalInfo<T, P>>,
         event_processor: ConcurrentEventProcessor<T, P>,
+        mut winning_proposals_sender: channel::Sender<ProposalInfo<T, P>>,
     ) {
         while let Some(proposal_info) = receiver.next().await {
             let guard = event_processor.read().compat().await.unwrap();
             match guard.process_proposal(proposal_info).await {
-                ProcessProposalResult::Done(_) => (),
+                ProcessProposalResult::Done(winning_proposal) => {
+                    if let Some(winning_proposal) = winning_proposal {
+                        if let Err(e) = winning_proposals_sender.send(winning_proposal).await {
```

Comment:
So what happens here? Are we concerned with the read lock being held while the task is waiting for the channel? I think it is fine but to be on the safe side we could keep this guard just for the duration of `process_proposal` (similarly to what we do in `sync_and_process_proposal`)